system/lib/wasmfs/syscalls.cpp

```diff
@@ -426,4 +430,79 @@ __wasi_errno_t __wasi_fd_seek(__wasi_fd_t fd,
 
   return __WASI_ERRNO_SUCCESS;
 }
+
+long __syscall_chdir(long path) {
+  auto pathParts = splitPath((char*)path);
+
+  if (pathParts.empty()) {
+    return -ENOENT;
+  }
+
+  long err;
+  auto dir = getDir(pathParts.begin(), pathParts.end(), err);
+
+  if (!dir) {
+    return err;
+  }
+
+  wasmFS.locked().setCWD(dir);
+  return 0;
+}
+
+long __syscall_getcwd(long buf, long size) {
+  // Check if buf points to a bad address.
+  if (!buf && size > 0) {
+    return -EFAULT;
+  }
+
+  // Check if the size argument is zero and buf is not a null pointer.
+  if (buf && size == 0) {
+    return -EINVAL;
+  }
+
+  auto curr = wasmFS.locked().getCWD();
+
+  std::string buffer = "";
```

Comment:
This isn't really a buffer in the sense of being a memory region into which we can write bytes, so how about calling it `result` instead?